src/app/pages/applications/forms/chart-form.component.ts

```diff
@@ -0,0 +1,343 @@
+import { Component } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import * as _ from 'lodash';
+import { Subscription } from 'rxjs';
+import { DialogService } from '../../../services/index';
+import { ApplicationsService } from '../applications.service';
+import { FieldConfig } from '../../common/entity/entity-form/models/field-config.interface';
+import { FieldSet } from '../../common/entity/entity-form/models/fieldset.interface';
+import { ModalService } from '../../../services/modal.service';
+import { EntityJobComponent } from '../../common/entity/entity-job/entity-job.component';
+import { CommonUtils } from 'app/core/classes/common-utils';
+import  helptext  from '../../../helptext/apps/apps';
+
+@Component({
+  selector: 'chart-form',
+  template: `<entity-form [conf]="this"></entity-form>`,
+})
+export class ChartFormComponent {
+  protected queryCall: string = 'chart.release.query';
+  protected queryCallOption: Array<any>;
+  protected addCall: string = 'chart.release.create';
+  protected editCall: string = 'chart.release.update';
+  protected isEntity: boolean = true;
+  protected utils: CommonUtils;
+
+  private title = helptext.nextCloudForm.title;
+  private name: string;
+  private getRow = new Subscription;
+  private rowName: string;
+  private dialogRef: any;
+  protected fieldConfig: FieldConfig[];
+  public fieldSets: FieldSet[] = [];
+  private catalogApp: any;
+
+  constructor(private mdDialog: MatDialog, private dialogService: DialogService,
+    private modalService: ModalService, private appService: ApplicationsService) {
+
+      this.getRow = this.modalService.getRow$.subscribe((rowName: string) => {
+        this.rowName = rowName;
+        this.queryCallOption = [["id", "=", rowName]];
+        this.getRow.unsubscribe();
+    })
+    this.utils = new CommonUtils();
+  }
+
+  getType(schema) {
+    let type;
+
+    switch (schema.type) {
+      case 'string':
+        type = 'input';
+        if (schema.enum) {
+          type = 'select';
+        }
+        break;
+      case 'boolean':
+        type = 'checkbox';
+        break;
+      case 'list':
+        type = 'list';
+        break;
+      case 'hostpath':
+        type = 'explorer';
+        break;
+      default:
+        type = 'input';
+    }
+
+    return type;
+  }
+
+  getSelectOptions(schema) {
+    const options = [];
+    schema.enum.forEach(option => {
+      options.push({
+        value: option.value,
+        label: option.description,
+      });
+    });
+    return options;
+  }
+
+  getFieldConfigs(config, parent=null) {
+    let fieldConfigs = [];
+    let name = config.variable;
+    if (parent) {
+      name = `${parent.variable}_${name}`;
+    }
+    const fieldConfig = {
+      type: this.getType(config.schema),
+      required: config.schema.required,
+      value: config.schema.default,
+      tooltip: config.description,
```

Comment:
A lot of tooltips and placeholders are also still hardcoded in:
https://github.com/freenas/webui/blob/master/src/app/helptext/apps/apps.ts